include/flatbuffers/flexbuffers.h

```diff
@@ -218,7 +218,8 @@ class String : public Sized {
 
   size_t length() const { return size(); }
   const char *c_str() const { return reinterpret_cast<const char *>(data_); }
-
+  std::string str() const { return c_str(); }
```

Comment:
thanks review!
i will follow the manager's decision about this PR.

my own test shows that the performance of std::string(c_str()) and std::string(c_str (), length()) were not significantly different. But the performance was better than about 3%.

```
auto start = chrono::high_resolution_clock::now();
for ( int i = 0 ; i < 100000000 ; ++ i)
{
    const char* str = "abcd";
    std::string s(str);
    // std::string s(str, strlen(str));
}
auto end = chrono::high_resolution_clock::now();
auto interval = chrono::duration_cast<chrono::milliseconds>(end - start);
cout << interval.count() << " msec" << endl;
```

i tested 10 times using release mode. (the unit is msec.)
string s(str) : 895, 887, 893, 927, 906, 894, 887, 903, 903, 901 (avg: 899.6)
string s(str, strlen(str)) : 946, 928, 933, 927, 928, 911, 936, 910, 947, 922 (avg: 928.8)

i will also follow the manager's decision.